### PR TITLE
configure datatables via html5 data attributes

### LINKED ISSUE #2154: Admin User Quota sorting is off
Thank you for opening an issue with Mailu. Please understand that issues are meant for bugs and enhancement-requests.
For **user-support questions**, reach out to us  on [matrix](https://matrix.to/#/#mailu:tedomum.net).

To be able to help you best, we need some more information.

## Before you open your issue
- [ x] Check if no issue or pull-request for this already exists.
- [ x] Check [documentation](https://mailu.io/master/) and [FAQ](https://mailu.io/master/faq.html). (Tip, use the search function on the documentation page)
- [ x] You understand `Mailu` is made by volunteers in their **free time** — be conscise, civil and accept that delays can occur.
- [ x] The title of the issue should be short and simple. It should contain specific terms related to the actual issue. Be specific while writing the title.

### Environment
 - [ x] docker-compose
 - [ ] kubernetes
 - [ ] docker swarm

### Versions
1.9

## Description
When sorting by quota in the Admin interface the numbers are sorted like text instead of by number and bytes.

## Expected behaviour
kB is smaller than MB is smaller than GB

quota

### PR DESCRIPTION
**Pull request by ghostwheel42**
## What type of PR?

bug-fix

## What does this PR do?

allows to sort most columns as a human would expect

### Related issue(s)
- closes #2154 

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [ ] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by ghostwheel42** [comment]
I'm still working on a good way to sort

 1)   relay targets (list of email addresses)
 2)  fetch tokens (ip addresses)
 3)   users/aliases (email addresses)

**Comment by Diman0** [comment]
> I'm still working on a good way to sort
> 
>     1. relay targets (list of email addresses)
> 
>     2. fetch tokens (ip addresses)
> 
>     3. users/aliases (email addresses)

I see that all those 3 points are altered by your PR. Does this mean you are happy with the sorting and can remove the WIP label?
If yes, please remove the label and I will review&test the PR.

**Comment by ghostwheel42** [comment]
I think merging this is an improvement even if ip- and email-addresses are sorted as string.
There are some glitches (list of signup_domain is not sortable and language selector bullet on the top right is missing), but this is for another issue.

**Comment by ghostwheel42** [comment]
bors try

**Comment by ghostwheel42** [comment]
fixed signup_domain list and language selector

**Comment by ghostwheel42** [comment]
bors try

File: core/admin/mailu/ui/templates/user/list.html [hunk 1]
```diff
@@ -39,14 +39,14 @@
...
+    <td data-ordered="{{ user.enable_imap*2 + user.enable_pop }}">
       {% if user.enable_imap %}<span class="badge bg-info">imap</span>{% endif %}
       {% if user.enable_pop %}<span class="badge bg-info">pop3</span>{% endif %}
     </td>
-    <td>{{ user.quota_bytes_used | filesizeformat }} / {{ (user.quota_bytes | filesizeformat) if user.quota_bytes else '∞' }}</td>
+    <td data-ordered="{{ user.quota_bytes_used }}">{{ user.quota_bytes_used | filesizeformat }} / {{ (user.quota_bytes | filesizeformat) if user.quota_bytes else '∞' }}</td>
     <td>{{ user.comment or '-' }}</td>
-    <td>{{ user.created_at | format_date }}</td>
-    <td>{{ user.updated_at | format_date }}</td>
+    <td data-ordered="{{ user.created_at or '0000-00-00' }}">{{ user.created_at | format_date }}</td>
```
Comment (Diman0) [inline core/admin/mailu/ui/templates/user/list.html:48]: When testing this PR, the sorting on this column works opposite as expected.
On the User List screen (https://hostname/admin/user/list/<domain>), I click on the Created column.

Ascending (Upward pointing arrow), the row with oldest date is on the top.
Descending (Downward pointing arrow), the row with newest date is on the top.

The Last Edit column works as expected.
image

**Review by Diman0 — COMMENTED** [review]
Can you explain this behaviour (see inline comment)?
I cannot explain this. Both columns use the some sorting but work differently.

File: core/admin/mailu/ui/templates/user/list.html (hunk 1 above)
Comment (ghostwheel42) [reply core/admin/mailu/ui/templates/user/list.html]: Is it working now?

**Comment by Diman0** [comment]
bors try

File: core/admin/mailu/ui/templates/user/list.html (hunk 1 above)
Comment (Diman0) [reply core/admin/mailu/ui/templates/user/list.html]: Yes it is!

**Review by Diman0 — APPROVED** [review]
LGTM

**Pull request merged**